Mathlib/ModelTheory/Basic.lean

```diff
@@ -137,7 +137,7 @@ theorem card_sum :
     (L.sum L').card = Cardinal.lift.{max u' v'} L.card + Cardinal.lift.{max u v} L'.card := by
   simp only [card_eq_card_functions_add_card_relations, card_functions_sum, card_relations_sum,
     sum_add_distrib', lift_add, lift_sum, lift_lift]
-  simp only [add_assoc, add_comm (Cardinal.sum fun i => (#(L'.Functions i)).lift)]
+  simp only [add_assoc, add_comm (Cardinal.sum fun i => Cardinal.lift #(L'.Functions i))]
```

Comment:
What's happening here?

Reply:
In Lean 4, dot notation doesn't work with structures like this.

Reply:
I'm not a fan of having both bundled and unbundled versions of structures - surely the more you can infer about a function from just the type signature, the better?

It's important to note this is one of only two places in Mathlib we were using dot notation for `lift` to begin with.

Reply:
> I'm not a fan of having both bundled and unbundled versions of structures - surely the more you can infer about a function from just the type signature, the better?

Well yes but in practice it only saves you one or two lemmas at the expense of your users needing to know whether the lemma holds true for a general bundled blah or just the specific one they have. Also, not applicable to your case, but there might not be a maximally general bundling.

Reply:
Of course, I understand you can't always get the ultimate bundling defined at the get go. In fact, `lift` should also be a ring hom, and I think it'd be nice to make a bundled version for that at some point.

I do think `InitialSeg` is a different situation - being an initial segment embedding is *precisely* the characteristic property of `lift`! All of the order properties of `lift`, like `lift_succ`, `lift_iSup`, ..., and even some other properties like `lift_lift`, follow from this fact, and I plan to generalize the proofs of these at some point.

Reply:
Please bring this discussion on Zulip

Reply:
I'll do this tomorrow, I'm a bit tired at the moment.

Reply:
I've made the thread at https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Cardinal.2Elift.20as.20an.20InitialSeg/near/471098568